resourcet/Control/Monad/Trans/Resource.hs

```diff
@@ -265,6 +265,10 @@ resourceForkWith g (ResourceT f) =
 --
 -- This is defined as @resourceForkWith forkIO@.
 --
+-- Note: Using regular 'forkIO' inside of a 'ResourceT' is inherently unsafe.
```

Comment:
Ideally this would end with ", since X" where X is a short, clear explanation of why it is unsafe, but I'm not familiar enough with resourceT to state that clearly

Reply:
The issue is that, if you use `forkIO`, the child thread isn't registered as using the internal state. Then, when the parent thread stops using the internal state, it will automatically clean up all of the resources still in the state, even though the child thread may be using them.

When you use `resourceForkIO` instead, it increments a counter to let the parent thread know that someone else is using the state. Only when the last thread finally stops using the step will the counter go down to 0 and the resources be cleaned up, when we know that no one is using them.

Reply:
Thanks, that's roughly what I thought, but wasn't positive. I updated the docs based on your explanation!

Reply:
![image](https://user-images.githubusercontent.com/1274145/83903893-cb955480-a713-11ea-855b-11ebe9ad2566.png)